tests/import/import_broken.py

```diff
@@ -0,0 +1,18 @@
+import sys
```

Comment:
We usually don't use `assert` in MicroPython tests. Instead, we `print` and the output is compared to the same from CPython. So I would suggest a test like this:

```python
import sys

print("sys" in sys.modules)

try:
    from broken import x
except Exception as e:
    print(e.__class__.__name__)

# The broken module we tried to import should not be in sys.modules.
print("broken" in sys.modules)

# If we try to import the module again, the code should
# run again and we should get the same error.
try:
    from broken import x
except Exception as e:
    print(e.__class__.__name__)
```

Reply:
The `print("sys" in sys.modules)` line reveals a bug in the current implementation as I mentioned in another comment. (It's a little bit surprising to me that we don't have a test that catches this already.)

Reply:
I misread your earlier message so that's why my response didn't really make sense.  You said `nlr_push` needs to be before the code that can raise the exception, and that is true.  The exception is raised within `do_load`, and every version of this pull request has called `nlr_push` before `do_load`.

The fact that `"sys"` is not in `sys.modules` is some other issue, which I'm not attempting to fix right now.  I added a test in `import_broken.py` that shows that normal, working modules are indeed added to `sys.modules`.

I went ahead and changed the code style the way you suggested though, putting the exception handling code *after* the code that throws the exception.  This requires about 27 lines of code to be indented, making it harder to fit code and comments in there, but I guess it is more readable.